core/dbt/config/runtime.py

```diff
@@ -328,10 +328,19 @@ def load_dependencies(self) -> Mapping[str, 'RuntimeConfig']:
         if self.dependencies is None:
             all_projects = {self.project_name: self}
             internal_packages = get_include_paths(self.credentials.type)
-            project_paths = itertools.chain(
-                internal_packages,
-                self._get_project_directories()
-            )
+            # raise exception if fewer installed packages than in packages.yml
+            package_dirs = self._get_project_directories()
+            count_packages_specified = len(self.packages.get('packages', 0))
+            count_packages_installed = len(tuple(package_dirs))
+            if count_packages_specified > count_packages_installed:
+                raise_compiler_error(
+                    f'dbt found {count_packages_specified} package(s) '
+                    f'specified in packages.yml, but only '
+                    f'{count_packages_installed} package(s) installed '
+                    f'in {self.modules_path}. Run "dbt deps" to '
+                    f'install package dependencies.'
+                )
+            project_paths = itertools.chain(internal_packages, package_dirs)
```

Comment:
...if we just need to call it again down here
```suggestion
            project_paths = itertools.chain(
                internal_packages,
                self._get_project_directories()
            )
```